aptos-move/framework/aptos-token/sources/listing.move

```diff
@@ -0,0 +1,243 @@
+module aptos_token::listing {
+    use std::signer;
+    use std::string::String;
+    use aptos_std::table::{Self, Table};
+    use aptos_token::token::{Self, TokenId};
+    use aptos_std::guid::{Self, ID};
+    use aptos_framework::account;
+    use aptos_framework::event::{Self, EventHandle};
+
+
+    const EOWNER_NOT_HAVING_ENOUGH_TOKEN: u64 = 1;
+    const ELISTING_NOT_EXIST:u64 = 2;
+
+    /// immutable struct for recording listing info.
+    struct Listing<phantom CoinType> has drop, store {
+        id: ID,
+        owner: address,
+        token_id: TokenId,
+        amount: u64,
+        min_price: u64,
+        instant_sale: bool, // true for marketplace and false for auction
+        start_sec: u64, // timestamp in secs for the listing starting time
+        expiration_sec: u64, // timestamp in secs for the listing expiration time
+    }
+
+    /// return a listing struct, marketplace owner can use this function to create a listing and store it in its inventory
+    public fun create_list<CoinType>(
```

Comment:
maybe `create_listing` or just `list`. the term as used implies you're creating a list data structure, e.g., a vector.

Reply:
why are we exposing this externally?

Reply:
This is for the use-case of launch-pad or large marketplace where they want to build their listing through partnering with some creators. The can store these listings in their deployed market_place contracts.